services/collaboration/pkg/connector/utf7/utf7.go

```diff
@@ -0,0 +1,308 @@
+package utf7
```

Comment:
Why can't we use an existing utf7 converter? Do we really need to maintain our own?

Reply:
The only encoder I see is [part of go-imap](https://github.com/mxk/go-imap/blob/master/imap/utf7.go). It does not follow the standard [Encoder / Decoder interfaces](https://pkg.go.dev/encoding#pkg-overview). Neither does this implementation. A good candidate to make a dedicated package? With its own Repo? And get the go-imap devs to collaborate on it?

But we won't be doing that ... because we are too busy doing our own shit :(

Reply:
As far as I know, the utf7 encoding "version" we need to use is the one defined in the https://datatracker.ietf.org/doc/html/rfc2152 . Basically, this means that the utf7-encoded string needs to be enclosed between "+" and "-".
There is an "imap" version of utf7 ("modified utf7" might be the right term) which uses "&" and "-" instead., which is what most of the libraries use because they're tailored for mailing.

I haven't seen any recent / maintained library implementing the RFC, so that's why I did my own..